### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by RitobrotoMukherjee**
## Performed review changes:
- Created `add_rental` method in both `Person` and `Book` objects to perform the missing `belongs-to` associations.

### Project requirements that I have performed:

- Create a class `Classroom` with the following:
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for `@label` (remember about `attr_accessor`).
- Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:
  - Create the `has-many` side (a classroom has many students).
  - Create the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class `Book` with the following:
  - `@title` and `@author` instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about `attr_accessor`).
- Create a class `Rental` with the following:
  - `@date` instance variable, should be initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).
- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.
- Added proper file structures
- **Added a file - `index.rb` to test the implements in a centralized manner. Commented out previous projects tests. 
To run it, need to move inside the project folder and run `ruby index.rb`**

### CODE REVIEW

File: classes/book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_reader :rentals
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (Lucas-Masaba) [inline classes/book.rb:9]: - [x] Please add an `add_rental` method that is similar to this below to create an association with rentals. 
```
  def add_rental(person, date)
    Rental.new(date, self, person)
  end

```

File: classes/person.rb [hunk 2]
```diff
@@ -11,6 +11,7 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @permission = parent_permission
+    @rentals = []
   end
```
Comment (Lucas-Masaba) [inline classes/person.rb:16]: - [x] Please add an `add_rental`  method to add an association to rentals. I can be something like this 👍 

```
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @RitobrotoMukherjee  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- Linter checks passing ✔️ 
- Clean code 🧹 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/book.rb (hunk 1 above)
Comment (RitobrotoMukherjee) [reply classes/book.rb:9]: Oh sure thank you for the suggestion. I am thinking to pass the whole rental object as parameters to `add_rental` function. Then I will not have to create the instance of `Rental` inside the method again

File: classes/person.rb (hunk 2 above)
Comment (RitobrotoMukherjee) [reply classes/person.rb:16]: Oh sure thank you for the suggestion. I am thinking to pass the whole rental object as parameter to `add_rental` method. Like this, I will not have to create an instance of `Rental` inside the method again.

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @RitobrotoMukherjee👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by RitobrotoMukherjee** [comment]
Hi @thecodechaser, Thank you for approving my project 🥳

**Pull request merged**